nuxeo-core/nuxeo-core-event/src/test/java/org/nuxeo/ecm/core/work/WorkManagerTest.java

```diff
@@ -254,6 +257,30 @@ public void testWorkManagerScheduling() throws Exception {
         assertWorkIdsEquals(Collections.emptyList(), null);
     }
 
+    @Test
+    @Deploy("org.nuxeo.ecm.core.event:test-work-failure-listeners.xml")
+    public void itCanFireWorkFailureEvent() throws Exception {
+        int initCount = DummyPostCommitEventListener.handledCount();
+        int initEvtCount = DummyPostCommitEventListener.eventCount();
+        int initSyncEvntCount = WorkFailureEventListener.getCount();
+
+        int duration = 2000; // 2s
+
+        SleepAndFailWork work = new SleepAndFailWork(0, false);
+        service.schedule(work);
+
+        assertTrue(service.awaitCompletion(duration * 3, TimeUnit.MILLISECONDS));
+
+        // synchronous listener
+        assertEquals(1 + initSyncEvntCount, WorkFailureEventListener.getCount());
+
+        EventService eventService = Framework.getService(EventService.class);
```

Comment:
Sorry I missed this the first time. You should `@Inject` the service instead of doing an explicit lookup with `Framework.getService`